source/exe/main.cc

```diff
@@ -32,12 +32,17 @@ int main(int argc, char** argv) {
 
 #ifdef ENVOY_HOT_RESTART
   // Enabled by default, except on OS X. Control with "bazel --define=hot_restart=disabled"
-  const std::string shared_mem_version = Envoy::Server::SharedMemory::version();
+  const Envoy::OptionsImpl::HotRestartVersionCB hot_restart_version_cb =
```

Comment:
nit: `HotRestartVersionCb`

Reply:
Or just `auto`, it should be inferable from its use in the call below.